asciidoc/courses/llm-knowledge-graph-construction/modules/3-python-create-graph/lessons/1-setup/lesson.adoc

```diff
@@ -0,0 +1,88 @@
+= Setup your development environment
+:order: 0
+:type: lesson
+:lab: {repository-link}
+:disable-cache: true
+
+In this module, you will use Python, Langchain, and OpenAI to create a knowledge graph from unstructured data.
+You must set up a development environment to run the code examples and exercises.
+
+We have link:{repository-link}[created a repository^] for this course.
+It contains the starter code and resources you need.
+
+== Get the code
+
+You can use link:https://gitpod.io[Gitpod^] as an online IDE and workspace for this workshop.
+It will automatically clone the workshop repository and set up your environment.
+
+lab::Open `Gitpod workspace`[]
+
+[NOTE]
+You will need to login with a Github, Gitlab, or Bitbucket account.
+
+Alternatively, you can clone the repository and set up the environment yourself.
+
+[%collapsible]
+.Develop on your local machine
+====
+You will need link:https://python.org[Python] installed and the ability to install packages using `pip`.
+
+You may want to set up a virtual environment using link:https://docs.python.org/3/library/venv.html[`venv`^] or link:https://virtualenv.pypa.io/en/latest/[`virtualenv`^] to keep your dependencies separate from other projects.
+
+Clone the link:{repository-link}[github.com/neo4j-graphacademy/llm-knowledge-graph] repository:
+
+[source,bash]
+----
+git clone https://github.com/neo4j-graphacademy/llm-knowledge-graph
+----
+
+Install the required packages using `pip` and download the required data:
+
+[source,bash]
+----
+cd llm-knowledge-graph
+pip install -r requirements.txt
+----
+====
+
+You do not need to create a Neo4j database as you will use the provided sandbox instance.
+
+You can find out more about how to configure Neo4j's GenAI functions in the link:https://neo4j.com/docs/cypher-manual/current/genai-integrations/[Neo4j GenAI integration documentation^].
+
+== Setup the environment
+
+Create a copy of the `.env.example` file and name it `.env`.
+Fill in the required values.
+
+[source]
+.Create a .env file
+----
+include::{repository-raw}/main/.env.example[]
+----
+
+Add your Open AI API key (`OPENAI_API_KEY`), which you can get from link:https://platform.openai.com[platformn.openai.com^].
+
+Update the Neo4j sandbox connection details:
+
+NEO4J_URI:: [copy]#bolt://{sandbox-ip}:{sandbox-boltPort}#
+NEO4J_USERNAME:: [copy]#{sandbox-username}#
+NEO4J_PASSWORD:: [copy]#{sandbox-password}#
+
+== Test your setup
+
+You can test your setup by running `llm-knowledge_graph\test_environment.py` - this will attempt to connect to the Neo4j sandbox and the OpenAI API.
+
+You will see an `OK` message if you have set up your environment correctly. If any tests fail, check the contents of the `.env` file.
```

Comment:
Can this be simplified into a unit test with a single command?  This will avoid any ambiguity with OS/slashes